File: 7KYU/ATM/solution.c
int nb_notes_needed (unsigned n)
{
  unsigned res = 0;
  
  while(n > 0) {
    if(n >= 500) {
      unsigned d = (unsigned)floor(n / 500);
      n -= 500 * d;
      res += d;
    } else if(n >= 200) {
      unsigned d = (unsigned)floor(n / 200);
      n -= 200 * d;
      res += d;
    } else if(n >= 100) {
      unsigned d = (unsigned)floor(n / 100);
      n -= 100 * d;
      res += d;
    } else if(n >= 50) {
      unsigned d = (unsigned)floor(n / 50);
      n -= 50 * d;
      res += d;
    } else if(n >= 20) {
      unsigned d = (unsigned)floor(n / 20);
      n -= 20 * d;
      res += d;
    }  else if(n >= 10) {
      unsigned d = (unsigned)floor(n / 10);
      n -= 10 * d;
      res += d;
    } else {
      return -1;
    }
  }
    
	return res;
}

File: 7KYU/ATM/solution.rb
def solve(n)
  res = 0
  while n > 0
    if n >= 500
      x = n / 500
      res += x
      n -= (x * 500)
    elsif n >= 200
      x = n / 200
      res += x
      n -= (x * 200)
    elsif n >= 100
      x = n / 100
      res += x
      n -= (x * 100)
    elsif n >= 50
      x = n / 50
      res += x
      n -= (x * 50)
    elsif n >= 20
      x = n / 20
      res += x
      n -= (x * 20)
    elsif n >= 10
      x = n / 10
      res += x
      n -= (x * 10)
    else
      return -1
    end
  end
  res
end
